pwndbg/aglib/macho.py

```diff
@@ -659,11 +660,30 @@ def shared_cache() -> DyldSharedCache | None:
     [1]: https://github.com/apple-oss-distributions/objc4/blob/f126469408dc82bd3f327217ae678fd0e6e3b37c/runtime/objc-opt.mm#L434
     [2]: https://github.com/apple-oss-distributions/dyld/blob/main/doc/dyld4.md#libdylddylib
     """
-    base = int(
-        pwndbg.dbg.selected_inferior().evaluate_expression(
-            "(const void*)_dyld_get_shared_cache_range()"
-        )
-    )
+    expression = pwndbg.dbg.selected_inferior().evaluate_expression
+
+    global global_weird_lldb_state
+    var = f"$_pwndbg_internal_shared_cache_size{global_weird_lldb_state}"
+
+    is_need_create = False
+    try:
+        expression(f"{var}=0; {var}")
+    except pwndbg.dbg_mod.Error as e:
+        if "Interpreter couldn't write to memory" in str(e):
+            # FIX for: https://github.com/llvm/llvm-project/issues/84806#issuecomment-1995055683
+            global_weird_lldb_state += 1
+            var = f"$_pwndbg_internal_shared_cache_size{global_weird_lldb_state}"
+            is_need_create = True
+        elif "use of undeclared identifier" in str(e):
+            is_need_create = True
+        else:
+            raise
+
+    if is_need_create:
+        expression(f"size_t {var}=0; {var}")
+
+    base = int(expression(f"(const void*)_dyld_get_shared_cache_range(&{var})"))
```

Comment:
Previously we did `_dyld_get_shared_cache_range()` now we do `_dyld_get_shared_cache_range(&somevar)`, is it ok we don't do the old version anymore?

Reply:
old version was wrong from beginning